File: 7kyu 2/isIsogram.js
/*
|--------------------------------------------------------------------------
| Isograms
|--------------------------------------------------------------------------
|
*/

function isIsogram(str) {
   const lower = str.toLowerCase();
   let letter = "";
   for (let i = 0; i < lower.length; i++) {
      if (letter.includes(lower[i]) === false) {
         letter += lower[i];
      } else {
         return false;
      }
   }
   return true;
}


function isIsogram(str) {
   return !/(\w).*\1/i.test(str)
}


function isIsogram(str) {
   const lower = str.toLowerCase()
   for (var i = 0; i < lower.length; i++) {
      if (lower.indexOf(lower.charAt(i), i + 1) >= 0) {
         return false
      }
   }
   return true
}


function isIsogram(str) {
   return new Set(str.toUpperCase()).size == str.length;
}


console.log(isIsogram("Dermatoglyphics"), true);
console.log(isIsogram("isogram"), true);
console.log(isIsogram("aba"), false, "same chars may not be adjacent");
console.log(isIsogram("moOse"), false, "same chars may not be same case");
console.log(isIsogram("isIsogram"), false);
console.log(isIsogram(""), true, "an empty string is a valid isogram");

File: Python/7py 2/is_isogram.py
"""
|--------------------------------------------------------------------------
| Isograms
|--------------------------------------------------------------------------
|
"""

def is_isogram(string):
   return len(set(list(string.lower()))) == len(string)


is_isogram = lambda _: len(set(_.lower()))==len(_)


def is_isogram(string):
   string = string.lower()
   for letter in string:
      if string.count(letter) > 1: return False
   return True


def is_isogram(string):
   return all(string.lower().count(i) <= 1 for i in string)


print(is_isogram("Dermatoglyphics"), True )
print(is_isogram("isogram"), True )
print(is_isogram("aba"), False, "same chars may not be adjacent" )
print(is_isogram("moOse"), False, "same chars may not be same case" )
print(is_isogram("isIsogram"), False )
print(is_isogram(""), True, "an empty string is a valid isogram" )